test/integration/action_client/active_job_assertions_test.rb

```diff
@@ -0,0 +1,226 @@
+        end
+      end
+    end
+
+    test "#assert_no_requests_enqueued raises without a block" do
+      request = MetricsClient.ping("status")
+
+      request.submit_later
+
+      assert_raises ActiveSupport::TestCase::Assertion do
+        assert_no_requests_enqueued
+      end
+    end
+
+    test "#assert_no_requests_enqueued raises from a block" do
+      request = MetricsClient.ping("status")
+
+      assert_raises ActiveSupport::TestCase::Assertion do
+        assert_no_requests_enqueued { request.submit_later }
+      end
+    end
+  end
+
+  class AssertRequestPerformedTest < ActiveJobAssertionsTestCase
+    test "#assert_performed_request accepts a request", rails: 6.0 do
+      ignore_http_requests!
+      perform_enqueued_jobs do
+        request = MetricsClient.ping("status")
+
+        request.submit_later
+
+        assert_performed_request request
+      end
+    end
+
+    test "#assert_performed_request accepts a block" do
+      ignore_http_requests!
+      request = MetricsClient.ping("status")
+
+      assert_performed_request(request) { request.submit_later }
+    end
+
+    test "#assert_performed_request accepts options", rails: 6.0 do
+      ignore_http_requests!
+      perform_enqueued_jobs do
+        request = MetricsClient.ping("status")
+        request.submit_later(queue: "low_priority")
+
+        assert_performed_request(request, queue: "low_priority")
+      end
+    end
+
+    test "#assert_enqueued_request raises with a request", rails: 6.0 do
+      perform_enqueued_jobs do
+        request = MetricsClient.ping("status")
+
+        assert_raises ActiveSupport::TestCase::Assertion do
+          assert_performed_request(request)
+        end
+      end
+    end
+
+    test "#assert_enqueued_request raises with a block" do
+      request = MetricsClient.ping("status")
+
+      assert_raises ActiveSupport::TestCase::Assertion do
+        assert_performed_request(request) {}
+      end
+    end
+
+    test "#assert_enqueued_request raises with options", rails: 6.0 do
+      ignore_http_requests!
+      perform_enqueued_jobs do
+        request = MetricsClient.ping("status")
+
+        request.submit_later(queue: "default")
+
+        exception = assert_raises(ActiveSupport::TestCase::Assertion) {
+          assert_performed_request(request, queue: "low_priority")
+        }
+        assert_includes exception.message, "low_priority"
+      end
+    end
+  end
+
+  class AssertNoPerformedRequestsTest < ActiveJobAssertionsTestCase
+    test "#assert_no_performed_requests does not fail when no requests are performed" do
+      assert_no_performed_requests
+    end
+
+    test "#assert_no_performed_requests accepts a block" do
+      assert_no_performed_requests { 1 + 1 }
+    end
+
+    test "#assert_no_performed_requests accepts a options", rails: 6.0 do
+      request = MetricsClient.ping("status")
+
+      assert_no_performed_requests(queue: "low_priority") do
```

Comment:
Should some of these `assert_no_performed_requests` tests be wrapped in `perform_enqueued_jobs do` to ensure we don't get a false positive?

Reply:
That's a good idea!